reposerver/repository/repository.go

```diff
@@ -794,6 +795,9 @@ func (s *Service) newClientResolveRevision(repo *v1alpha1.Repository, revision s
 	if err != nil {
 		return nil, "", err
 	}
+	if git.IsCommitSHA(revision) {
```

Comment:
The same check is in LsRemote method: https://github.com/argoproj/argo-cd/blob/bbfb96cb0161b1d11cbbc334fb33eb1158b2c075/util/git/client.go#L334 . So this check is not required